### ISSUE TITLE
Contractive Types and isVoid()

### ISSUE THREAD

**Issue by DavePearce** [Whiley/WhileyCompiler#885]
_(related to #884)_

There is basically a problem with `StrictTypeEmptinessTest.isVoid()` which, in the presence of non-contractive types, goes into an infinite loop.  For example, this type:

```
type Contractive is Contractive | int
```

Under the old approach to reporting syntax errors, this was caught using the method `isContractive()` and then nothing happened after that.  Under the new regime, however, we continue type checking and, hence, even though `isContractive()` fails we still continue with `isVoid()`.
